common4j/src/test/com/microsoft/identity/common/java/providers/microsoft/MockMicrosoftAuthorizationRequest.java

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.java.providers.microsoft;
+
+public class MockMicrosoftAuthorizationRequest extends MicrosoftAuthorizationRequest<MockMicrosoftAuthorizationRequest> {
+
+    public static final String MOCK_AUTH_ENDPOINT = "https://MOCK_AUTH_ENDPOINT";
+
+    /**
+     * Constructor of MockMicrosoftAuthorizationRequest.
+     *
+     * @param builder
+     */
+    protected MockMicrosoftAuthorizationRequest(Builder builder) {
+        super(builder);
+    }
+
+    @Override
+    public String getAuthorizationEndpoint() {
+        return MOCK_AUTH_ENDPOINT;
+    }
```

Comment:
This class and it's sibling seem to exist only to override this endpoint.  Can we just set it on the non-mocked class, or am I missing something important?

Reply:
The non-mocked class (I assume you mean MicrosoftAuthorizationRequest itself) is an abstract class.